Build error "lua: ./h/em_table_lib.lua:38: malformed pseudo line in em_table", on Ubuntu 22.04 + Lua 5.1.5, & segfault on Ubuntu 22.04 + Lua 5.2.4

Hello @davidgiven,

I am getting a fatal error trying to build ACK on Ubuntu 22.04 LTS with Lua 5.1.5:

```
$ make BUILDDIR="..." PREFIX='$(INSDIR)'
...
[25/10116] lua modules/src/em_code/mak.../src/em_code/em_code_ek_h/em_codeEK.h 
FAILED: .../obj/modules/src/em_code/em_code_ek_h/em_codeEK.h 
lua modules/src/em_code/make_codeek_h.lua < h/em_table > .../obj/modules/src/em_code/em_code_ek_h/em_codeEK.h && cat modules/src/em_code/em.nogen >> .../obj/modules/src/em_code/em_code_ek_h/em_codeEK.h 
lua: ./h/em_table_lib.lua:38: malformed pseudo line in em_table
stack traceback:
	[C]: in function 'error'
	./h/em_table_lib.lua:38: in function 'load_table'
	modules/src/em_code/make_codeek_h.lua:3: in main chunk
	[C]: ?
[26/10116] lua modules/src/em_data/mak...dules/src/em_data/em_flag_c/em_flag.c 
FAILED: .../obj/modules/src/em_data/em_flag_c/em_flag.c 
lua modules/src/em_data/make_flag_c.lua < h/em_table > .../obj/modules/src/em_data/em_flag_c/em_flag.c 
lua: ./h/em_table_lib.lua:38: malformed pseudo line in em_table
stack traceback:
	[C]: in function 'error'
	./h/em_table_lib.lua:38: in function 'load_table'
	modules/src/em_data/make_flag_c.lua:3: in main chunk
	[C]: ?
...
$ lua -v
Lua 5.1.5  Copyright (C) 1994-2012 Lua.org, PUC-Rio
```

The error seems to be caused by the lines
```
		local _, _, name, num, flags = line:find("^(%w+)%s+(%d+)%s+(%g+)$")
		if not name then
			error("malformed pseudo line in em_table")
		end
```
in `h/em_table_lib.lua` while trying to parse
```
bss	0	nvt
```
in `h/em_table`.

Thank you!

Hello @davidgiven,

It seems that Lua 5.1 does not support the `%g` character class that `h/em_table_lib.lua` uses (though Lua 5.2 and above does support it).

Incidentally, the build (mostly) succeeds if I upgrade to Lua 5.2.4 or Lua 5.4.4, though it now crashes further down:

```
[4051/10116] ACKDIR=.../obj/lang/m2/include/headers -O6 
FAILED: .../obj/lang/m2/libm2/lib_cpm/Storage/Storage.o 
ACKDIR=.../staging .../staging/bin/ack -mcpm -c.o -o .../obj/lang/m2/libm2/lib_cpm/Storage/Storage.o lang/m2/libm2/Storage.mod -I.../obj/h/emheaders -I.../obj/lang/m2/include/headers -O6 
Segmentation fault (core dumped)
[4056/10116] ACKDIR=.../obj/lang/m2/include/headers -O6 
ninja: build stopped: subcommand failed.
```

Thank you!

--- a/lang/m2/comp/scope.c
+++ b/lang/m2/comp/scope.c
@@ -112,10 +112,11 @@ static void chk_forw(struct def **pdf)
 	register struct def *df;
 
 	while ( (df = *pdf) ) {
-		if (df->df_kind == D_FORWTYPE) {
-			pdf = &df->df_nextinscope;
+		while (df->df_kind == D_FORWTYPE) {
+			register struct def *df2 = df->df_nextinscope;
+			pdf = NULL;
 			ForceForwardTypeDef(df);	/* removes df */
-			continue;
+			df = df2;
 		}
 		if (df->df_kind & (D_FORWARD|D_FORWMODULE)) {
 			/* These definitions must be found in
@@ -138,7 +139,8 @@ df->df_idf->id_text);
 						nextvisible(CurrVis);
 				register struct def *df1 = lookup(df->df_idf, ls->sc_scope, 0, 0);
 
-				*pdf = df->df_nextinscope;
+				if (pdf)
+					*pdf = df->df_nextinscope;
 	
 				if (! df1) {
 					if (df->df_kind == D_FORWMODULE) {

